internal/getter/file.go

```diff
@@ -1,15 +1,48 @@
 package getter
 
 import (
+	"fmt"
 	"path/filepath"
+	"runtime"
 
 	"github.com/hashicorp/go-getter"
 	"github.com/spf13/afero"
 )
 
 // fileDetector is a replacement for go-getter's own file detector which also verifies the source file exists
 type fileDetector struct {
-	fd getter.FileDetector
+	fd               getter.FileDetector
+	allowNonExisting bool
+}
+
+func isValidPath(fs afero.Fs, path string) bool {
+	err := fs.MkdirAll(path, 0700)
```

Comment:
This doesn't check for validity but for write access also. Why not fs.Stat first, if `os.ErrNotExist` just use methods from `filepath` to make sure it's a valid syntax? (or just check syntax and don't do any stat)

Creating just to remove doesn't shouldn't even be an option.

Reply:
We do an `Exists` check before https://github.com/cloudquery/cloudquery/blob/8520eae35d47a89a86406a8e36831f7903edec52/internal/getter/file.go#L23 so I'm not sure we need another one here.

Checking for valid syntax is not straightforward as it's highly dependent on OS flavor and version (e.g. OSX, Windows, Linux and even different version within that OS).

We could use `WriteFile` maybe but that will fail if the directory structure is missing (however maybe we can detect that from the error).

More in https://stackoverflow.com/questions/35231846/golang-check-if-string-is-valid-path